src/tigerbeetle/main.zig

```diff
@@ -227,6 +227,34 @@ const Command = struct {
                 "if it's unexpected, please recompile TigerBeetle with -Dconfig-aof-recovery=false.", .{replica.replica});
         }
 
+        // It is possible to start tigerbetele passing `0` as an address:
+        //     $ tigerbeetle start --addresses=0 0_0.tigrebeetle
+        // This enables a couple of special behaviors, useful in tests:
+        // - The operating system picks a free port, avoiding "address already in use" errors.
+        // - The port, and only the port, is printed to the stdout, so that the parent process
+        //   can learn it.
+        // - tigerbeetle process exits when its stdin gets closed.
+        const port_zero = std.net.Address.parseIp(constants.address, 0) catch unreachable;
+        if (args.addresses[replica.replica].eql(port_zero)) {
```

Comment:
Maybe we should make this a bit more explicit?

If `constants.address` is eg `127.0.0.1`, then:

`--addresses=0` will trigger this mode, as will `--addresses=127.0.0.1:0` but not `--addresses=127.0.0.2:0`

Reply:
Changed to check for literal `"0"` to lock this down. Enabling magic-zero mode for clusters of more than one machine seems not that useful, and, indeed, it's better to lock down anything we are not specifcially designing for!